### PR TITLE
extensibility: refresh diff status bars

### LINKED ISSUE #21716: Extension buttons on diffs need redesign work
Full height extension buttons in the diff views (and other areas where these elements exist) should be redesigned. 

For example, the buttons in the panel at the top of diff views should be links instead. Figma:

https://www.figma.com/file/30vUPTo5zeoQXdj5c6478w/Extension-buttons-in-diffs?node-id=0%3A1&viewport=485%2C404%2C0.31086957454681396

### PR DESCRIPTION
**Pull request by tjkandala**
Closes #21716. 

Screenshot from 2021-06-16 08-36-24
Screenshot from 2021-06-16 08-36-59

Only show head status bar in unified mode, show both in split mode. Remove hover styles. @AlicjaSuska should we have a different text color/underline on hover?

<details>
<summary>Split status bars scroll when overflowing</summary>

</details>

<details>
<summary>No more full height background on hover</summary>

</details>

<details>
<summary>Toggle between split and unified mode</summary>

</details>

TODO (based on latest information):
- [ ] Show base and head status bars on unified mode on separate lines

### CODE REVIEW

File: client/web/src/components/diff/FileDiffHunks.tsx [hunk 1]
```diff
@@ -124,43 +125,36 @@ export const FileDiffHunks: React.FunctionComponent<FileHunksProps> = ({
...
-                switchMap(([baseViewerId, headViewerId, extensionHostAPI]) =>
-                    combineLatest([
-                        baseViewerId
-                            ? wrapRemoteObservable(extensionHostAPI.getStatusBarItems(baseViewerId))
-                            : of(null),
-                        headViewerId
-                            ? wrapRemoteObservable(extensionHostAPI.getStatusBarItems(headViewerId))
-                            : of(null),
-                    ])
+                distinctUntilKeyChanged('headViewerId'),
```
Comment (vovakulikov) [inline client/web/src/components/diff/FileDiffHunks.tsx:141]: **Q:** Is it here because we don't want to react on baseViewerID anymore?

**Review by vovakulikov — APPROVED** [review]
LGTM. I like this comprehensive description!

File: client/web/src/components/diff/FileDiffHunks.tsx (hunk 1 above)
Comment (tjkandala) [reply client/web/src/components/diff/FileDiffHunks.tsx:141]: Yep, technically each viewer ID should only change once (from `undefined` to fixed `ViewerId`), but the base and head viewer IDs could be returned from the extension host at different times, and we don't want to unsub and resub to `extensionHostAPI.getStatusBarItems(viewerId)`, especially with the [long ritual/song and dance we have to do to observe values across contexts](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/client/shared/src/api/client/api/common.ts?L41-51). There's a lot of Web Worker overhead on the commit page :)

**Review by AlicjaSuska — CHANGES_REQUESTED** [review]
Thank you for implementing this @tjkandala. Here are things that I think we should change:
- in the unified view, we should display head and base labels as well. I've commented in this [message](https://sourcegraph.slack.com/archives/C01T0BL4EFN/p1623788219032300?thread_ts=1623177353.000700&cid=C01T0BL4EFN) (I've seen you gave the thumbs up)
- please, reduce space between the extension static items, it should be 8px now it's 12px (4px tight padding, 4px margin, 4 px left padding)
- In [Figma](https://www.figma.com/file/30vUPTo5zeoQXdj5c6478w/?node-id=0%3A1), clickable elements have the link styles. Can you make this change? It also answers the question about the hover state (the same as for links: underline)

Thank you @tjkandala! Could you please ping me / request a review again when the changes are ready? :) 


Spacing screenshots:
Screenshot 2021-06-18 at 09 35 02
Screenshot 2021-06-18 at 09 34 52

**Comment by tjkandala** [comment]
Hi @AlicjaSuska, I made the changes you requested:

Screenshot from 2021-07-12 11-28-38
Screenshot from 2021-07-12 11-28-45

We have to apply link styles for the whole status bar item from an extension (if clickable) since we can't determine which part is semantically the "link," unlike the design in which `Author: ` didn't have link styles.

**Review by AlicjaSuska — APPROVED** [review]
Thank you @tjkandala ! From the screenshots, it looks good :)

*I couldn't see it locally though. Changes to spacing and link color weren't showing up, even thought I've pulled changes from this branch.

**Pull request merged**